Have the counter follow the mouse

Manlet008 opened the pull request:
Gives the option to have a counter follow the mouse, as well as options to update the x, y offset.

This counter appears above the ui.

Manlet008 committed:
Uhh mouse tracking

Manlet008 committed:
Moved the tiemr to appear above the ui

Manlet008 committed:
The tick counter will follow the mouse even when the mouse is clicked and dragging

vincent0955 commented on src/main/java/com/visualmetronome/VisualMetronomeConfig.java:
```diff
@@ -23,6 +23,49 @@ default boolean enableMetronome()
+		return false;
+	}
+	@ConfigItem(
+			position = 1, // Choose appropriate position
```
Can you get rid of all comments like these in the pull request

Manlet008 replied:
Done

vincent0955 commented on src/main/java/com/visualmetronome/MouseFollowingOverlay.java:
```diff
@@ -0,0 +1,60 @@
+                );
+
+                // Render
+                OverlayUtil.renderTextLocation(graphics, textPosition, text, config.NumberColor());
```
If you could render the other color option if the config option is checked that would be good

eg. in VisualMetronomeNumberOverlay we have this
```
// Set which color to use
            Color numberColor;
            if (config.overheadUseCurrentColor())
            {
                numberColor = plugin.currentColor;
            }
            else
            {
                numberColor = config.NumberColor();
            }
```

Manlet008 replied:
Done

vincent0955 commented on src/main/java/com/visualmetronome/VisualMetronomeConfig.java:
```diff
@@ -143,6 +186,14 @@ default Color NumberColor()
+	@ConfigSection(
+			name = "Mouse Following Settings",
+			description = "Settings for mouse-following tick counter",
+			position = 6, // Choose appropriate position
```
There is already a section in position 6, set this position to 9 and move the next section to position 10

Manlet008 replied:
Done

vincent0955 commented on src/main/java/com/visualmetronome/MouseFollowingOverlay.java:
```diff
@@ -0,0 +1,60 @@
+import net.runelite.api.Point;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.overlay.*;
```
Just import the libraries you need, don't import all like this

Manlet008 replied:
Done

vincent0955 commented on src/main/java/com/visualmetronome/MouseFollowingOverlay.java:
```diff
@@ -0,0 +1,60 @@
+import net.runelite.client.ui.overlay.*;
+
+import javax.inject.Inject;
+import java.awt.*;
```
Same with this one

Manlet008 replied:
Done

Manlet008 committed:
- Comments removed from Config
- Mouse Following Counter now changes colours
- Mouse Following Settings moved to section 9, other second moved to 10
- Only needed libraries are now being imported.

vincent0955 committed:
Formatting, removing some comments

vincent0955 commented:
I've made some changes to your fork to fix some formatting while reviewing again, but I noticed a bug that when you have the option checked off, and you restart the client, it disappears, and you have to reenable it. Can you fix this? 

Another thing is I don't know if you can fix the lag, but the number render is a little laggy and makes my mouse movement feel more sluggish, even if it's just the number thats lagging. I don't know how viable this would be for actual users, because it definitely bother me a bit when I use it.

Manlet008 committed:
The mouse tracker now works properly on start up if it's already enabled in the config

Manlet008 committed:
Merge remote-tracking branch 'origin/v2' into v2

# Conflicts:
#	src/main/java/com/visualmetronome/VisualMetronomePlugin.java

Manlet008 committed:
Revert "Merge remote-tracking branch 'origin/v2' into v2"

This reverts commit 59657c5782815e908854d1bdee76349ab91a6a0d, reversing
changes made to e25842be8e95e1d84624b1bc3c40dc0016ea0ca6.

vincent0955 committed:
Reset to commit d60920d4b831cc28cecbdfafc707178e99a593aa

vincent0955 committed:
Readded mouse tracker works properly on start up changes
